src/ioctls/vm.rs

```diff
@@ -1799,6 +1799,38 @@ mod tests {
         assert!(vm.set_gsi_routing(&irq_routing).is_ok());
     }
 
+    #[test]
+    fn create_vcpu_different_cpuids() {
+        let kvm = Kvm::new().unwrap();
+        let vm = kvm.create_vm().unwrap();
+
+        // Fails when an arbitrarily large value
+        let err = vm.create_vcpu(65537 as u64).err();
+        assert_eq!(err.unwrap().errno(), libc::EINVAL);
+
+        // Note: We can request up to KVM_MAX_VCPU_ID if it exists or up to KVM_MAX_VCPUS or
+        // NR_CPUS or 4. This is determined by the appropriate capability being present.
+        // We check near boundry conditions `max_vcpus - 1` should succeed but `max_vcpus` as
+        // determined by the appropriate capability should fail.
+        //
+        // Ref: https://www.kernel.org/doc/html/latest/virt/kvm/api.html#kvm-create-vcpu
+        //
+        let mut max_vcpus = vm.check_extension_int(Cap::MaxVcpuId);
+        if max_vcpus == 0 {
+            max_vcpus = vm.check_extension_int(Cap::MaxVcpus);
+        }
+        if max_vcpus == 0 {
+            max_vcpus = vm.check_extension_int(Cap::NrVcpus);
+        }
+        if max_vcpus == 0 {
+            max_vcpus = 4
+        }
```

Comment:
I think, this should be made a separate private function something like `vm.get_max_vcpu_id`? This is perhaps useful outside this test case as well.

Reply:
yes, I remember there was such a code somewhere, but I can't seem to remember where.

Reply:
Ok, I got it now. You can use the `get_max_vcpus` function that already exists in the Kvm interface.

Reply:
`get_max_vcpus` and `get_max_vcpu_id` are different ioctls as per the [documentation here](https://www.kernel.org/doc/html/latest/virt/kvm/api.html#kvm-create-vcpu) . For instance when I checked on `x86_64` on my laptop, `max_vcpus` was 288 but `max_vcpu_id` was 1023. So I believe a function like `get_max_vcpu_id` should be added to `kvm` interface? 

and then that can be used in this test case?

Reply:
Yap, that sounds like a good idea.